File: csla20cs/Csla/Web/Design/ObjectSchema.cs
using System;
using System.Data;
using System.Web.UI;
using System.Web.UI.Design;
using System.ComponentModel;
using System.Reflection;
using System.Collections;
using System.Collections.Generic;

namespace Csla.Web.Design
{
  /// <summary>
  /// Object providing access to schema information for
  /// a business object.
  /// </summary>
  /// <remarks>
  /// This object returns only one view, which corresponds
  /// to the business object used by data binding.
  /// </remarks>
  public class ObjectSchema : IDataSourceSchema
  {

    private string _typeAssemblyName = string.Empty;
    private string _typeName = string.Empty;

    /// <summary>
    /// Creates an instance of the object.
    /// </summary>
    /// <param name="assemblyName">Assembly containing
    /// the type.</param>
    /// <param name="typeName">Type name for
    /// which the schema should be generated.</param>
    public ObjectSchema(string assemblyName, string typeName)
    {
      _typeAssemblyName = assemblyName;
      _typeName = typeName;
    }

    /// <summary>
    /// Returns a single element array containing the
    /// schema for the CSLA .NET business object.
    /// </summary>
    public IDataSourceViewSchema[] GetViews()
    {
      IDataSourceViewSchema[] result;
      result = new IDataSourceViewSchema[] 
        { new ObjectViewSchema(_typeAssemblyName, _typeName) };
      return result;
    }

  }
}

File: csla20vb/Csla/Web/Design/ObjectSchema.vb
Imports System.Web.UI
Imports System.Web.UI.Design
Imports System.ComponentModel
Imports System.Reflection

Namespace Web.Design

  ''' <summary>
  ''' Object providing access to schema information for
  ''' a business object.
  ''' </summary>
  ''' <remarks>
  ''' This object returns only one view, which corresponds
  ''' to the business object used by data binding.
  ''' </remarks>
  Public Class ObjectSchema
    Implements IDataSourceSchema

    Private mTypeName As String = ""
    Private mDesigner As CslaDataSourceDesigner

    ''' <summary>
    ''' Creates an instance of the object.
    ''' </summary>
    ''' <param name="typeName">Type name for
    ''' which the schema should be generated.</param>
    Public Sub New(ByVal designer As CslaDataSourceDesigner, ByVal typeName As String)

      mTypeName = typeName
      mDesigner = designer

    End Sub


    ''' <summary>
    ''' Returns a single element array containing the
    ''' schema for the CSLA .NET business object.
    ''' </summary>
    Public Function GetViews() As _
      System.Web.UI.Design.IDataSourceViewSchema() _
      Implements System.Web.UI.Design.IDataSourceSchema.GetViews

      Dim result As IDataSourceViewSchema()
      result = New IDataSourceViewSchema() _
        {New ObjectViewSchema(mDesigner, mTypeName)}
      Return result

    End Function

  End Class

End Namespace